build-settings/rollup.config.js

```diff
@@ -37,49 +60,76 @@ const createSharedConfig = (pkgName) => ({
- * have the benefits of ES6 like tree shaking. In reality, the CJS output is
- * likely only used by Jest anyway. We could force consumers to use Babel around
- * their imports so that Jest mocking is supported, but that feels mean and
- * messy.
- */
-const createConfig = (pkgName) => {
-    const sharedConfig = createSharedConfig(pkgName);
+// For each package in our packages folder, generate the outputs we want.
+// To determine what those outputs are, we read the package.json file for
+// each package. If the package has a `browser` field, then we generate
+// browser and node assets. If not, we just generate the node assets.
+// Note that we also get the output paths from the package.json.
+const getPackageInfo = (pkgName) => {
+    const pkgJsonPath = makePackageBasedPath(pkgName, "./package.json");
+    const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath));
+
+    // Now we have the package.json, we need to look at the main, module, and
+    // browser fields and values.
+    const {main: cjsNode, module: esmNode, browser} = pkgJson;
+    const cjsBrowser = browser == null ? undefined : browser[cjsNode];
+    const esmBrowser = browser == null ? undefined : browser[esmNode];
+
     return [
         {
-            output: {
-                file: `packages/${pkgName}/dist/index.js`,
-                sourcemap: true,
-                format: "cjs",
-            },
-            ...sharedConfig,
+            name: pkgName,
+            format: "esm",
+            platform: "node",
+            file: esmNode,
         },
         {
-            output: {
-                file: `packages/${pkgName}/dist/es/index.js`,
-                sourcemap: true,
-                format: "esm",
-            },
-            ...sharedConfig,
+            name: pkgName,
+            format: "esm",
+            platform: "browser",
+            file: esmBrowser,
+        },
+        {
+            name: pkgName,
+            format: "cjs",
+            platform: "node",
+            file: cjsNode,
         },
-    ];
+        {
+            name: pkgName,
+            format: "cjs",
+            platform: "browser",
+            file: cjsBrowser,
+        },
+    ].filter((i) => !!i.file);
 };
 
-// For each package in our packages folder, generate the outputs we want.
-export default fs.readdirSync("packages").flatMap(createConfig);
+/**
+ * Creates the full rollup configuration for the given args.
+ *
+ * If the `configPackages` arg is included, we split it on commas and
+ * take each as the name of a package to process. Otherwise, we process all
+ * packages.
+ */
+const createRollupConfig = (commandLineArgs) => {
+    const {configPackages} = commandLineArgs;
+    const pkgNames =
+        configPackages && configPackages.length
+            ? configPackages
+                  .split(",")
+                  .map((s) =>
+                      s.startsWith("wonder-stuff") ? s : `wonder-stuff-${s}`,
```

Comment:
This bit means we can target rollup at specific packages if we want.

Reply:
Is there any way to make this generic? (e.g. not reference the string `wonder-stuff` directly?) If so we could re-use this rollup config in a bunch of other places, including Wonder Blocks and maybe Perseus!

Reply:
Ideally I'd like to avoid copy-and-pasting configs as it will mean that things will get quickly out-of-sync, so anything we can do to make it generic and easily shareable would be ideal!